File: trunk/theLink/example/cc/mulclient.cc
/**
 *  \file       theLink/example/cc/mulclient.cc
 *  \brief      \$Id$
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev$
 *  \author     EMail: aotto1968 at users.berlios.de
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */

#include <iostream>
#include "ccmsgque.h"

using namespace std;
using namespace ccmsgque;

int MQ_CDECL main (int argc, MQ_CST argv[])
{
  MqC ctx;
  try {
    ctx.ConfigSetName ("MyMulClient");
    ctx.LinkCreateVC (argc, argv);
    ctx.SendSTART();
    ctx.SendD(3.67);
    ctx.SendD(22.3);
    ctx.SendEND_AND_WAIT("MMUL", 5);
    cout << ctx.ReadD() << endl;
  } catch (const exception& e) {
    ctx.ErrorSet(e);
  }
  ctx.Exit();
}

File: tags/NHI1-0.5/theLink/example/vb/mulclient.vb
REM
REM  \file       theLink/example/vb/mulclient.vb
REM  \brief      \$Id$
REM  
REM  (C) 2009 - NHI - #1 - Project - Group
REM  
REM  \version    \$Rev$
REM  \author     EMail: aotto1968 at users.berlios.de
REM  \attention  this software has GPL permissions to copy
REM              please contact AUTHORS for additional information
REM

Imports System
Imports csmsgque

Public Module example
  Sub Main(ByVal args() As String)
    Dim ctx As New MqS()
    Try
      ctx.LinkCreate(args)
      ctx.SendSTART()
      ctx.SendD(3.67)
      ctx.SendD(22.3)
      ctx.SendEND_AND_WAIT("MMUL", 5)
      Console.WriteLine(ctx.ReadD())
    Catch ex As Exception
      ctx.ErrorSet(ex)
    Finally
      ctx.Exit()
    End Try
  End Sub
End Module
